pennylane/workflow/qnode.py

```diff
@@ -742,6 +745,14 @@ def circuit(x):
             raise ValueError(
                 f"Must specify at least one configuration property to update. Valid properties are: {valid_params}."
             )
+
+        # For shots, we allow `shots=...` to be passed as a keyword argument
+        # and this should happen before any other updates (device also has shots, until they were deprecated)
+        shots = self._shots
+        shots_to_update = "shots" in kwargs
+        if shots_to_update:
+            shots = qml.measurements.Shots(kwargs.pop("shots"))
+
```

Comment:
Since we now have `update_shots`, I don't think we need to do this is `update` anymore.